eng/Versions.props

```diff
@@ -154,6 +154,31 @@
     -->
     <MicrosoftNETCoreAppRuntimeVersion>$(MicrosoftNETCoreAppRuntimewinx64PackageVersion)</MicrosoftNETCoreAppRuntimeVersion>
   </PropertyGroup>
+
+  <!--
+    We ship ref/ assemblies for runtime packages in our targeting targeting pack. When servicing that targeting pack,
+    these assemblies must no change. Must also compile our shared framework and ref/ assemblies against the initial ref/
+    assemblies for runtime packages. But, need to test against the latest implementation assemblies and ship them in our
+    shared framework. Upshot is we need Major.Minor.0 runtime packages for compilation and the targeting pack and
+    Major.Minor.Latest runtime packages for everything else.
+    !!! List below is likely incomplete. Should include all dotnet/runtime packages containing assemblies in
+    !!! Microsoft.AspNetCore.App.Ref or Microsoft.AspNetCore.App.Runtime.*.
+  -->
+  <PropertyGroup Condition=" '$(IsServicingBuild)' == 'true' ">
```

Comment:
Why did you remove `AND '$(MSBuildProjectName)' == 'Microsoft.AspNetCore.App.Ref'` from the condition? We still want to reference the latest everywhere _except_ the ref pack

Reply:
Because the condition was incorrect in 'master'. The 'release/3.1' condition is closer to what's needed: `$(MSBuildProjectName)' == 'Microsoft.AspNetCore.App.Ref' OR '$(IsReferenceAssemblyProject)' == 'true'`. But, that's incomplete because we split the ref/ and implementation versions in that branch.

I could conditionally compile implementation projects against the assemblies in these packages and test against the latest versions. But, I'm not sure that's worthwhile. Thoughts?

Reply:
Why would this not be worthwhile?

Reply:
It's worthwhile We discussed this offline and need to invoke the gods: @Pilchie and @ericstj 😺 

We had issues in the past because the updated packages from dotnet/corefx and dotnet/core-setup contained ref/ assemblies with newer assembly versions. This meant the ref/ assemblies built in dotnet/aspnetcore _required_ the newer ref/ assemblies when compiling and that wasn't our intent. @ericstj will newer ref/ assemblies change assembly versions in 5.0 servicing of dotnet/runtime❔

@Pilchie please confirm our requirements here are to
1. include the 5.0.0 ref/ assemblies from dotnet/runtime in our targeting packs
2. include any updated 5.0.x implementation assemblies from dotnet/runtime  in our runtime packs
3. compile against the 5.0.0 ref/ assemblies from dotnet/runtime
4. test against the 5.0.x implementation assemblies from dotnet/runtime

If that's correct, I can try referencing both packages, using `Compile` assets from the 5.0.0 packages and everything else from newer 5.0.x packages. But, I'm doubtful that will work correctly up the project dependency tree in this repo. I'm also not sure how to test this works correctly without dotnet/runtime and us doing a servicing exercise. Suggestions❔

Reply:
Those requirements seem correct from me.

Reply:
It seems like it should be possible/supported (by roslyn) to always build against the oldest and run against the latest - maybe @agocke or someone else from Roslyn knows how we can accomplish that?

Reply:
That's not a question of the compiler, it's a question of the package contents, and msbuild targets that locate things.  The compiler just gets a path on disk to the thing to reference.

Reply:
> @ericstj will newer ref/ assemblies change assembly versions in 5.0 servicing of dotnet/runtime

Ref assemblies that are part of the ref-pack will never change in servicing.  Ref assemblies in the transport package we provide for extensions should also not change.

Individual packages will change as needed so that they can have unique assembly versions.

Reply:
🆗 @ericstj that means I need to hack things to get our projects compiling against 5.0.0 versions of the assemblies and running against 5.0.x for this specific set of packages. I haven't tried my "reference both versions" idea yet and still have no idea how to test it out without a dotnet/runtime / dotnet/aspnetcore servicing exercise.

Reply:
Wasn’t this what was already happening in 3.x?  I thought that’s exactly what we made work for the ref pack servicing.  Let me know if there is something different we should be doing to make things easier.

Reply:
Yes, we're dealing with similar incoming packages in 3.x but aren't using Roslyn to generate the ref/ assemblies. That means it's _much_ easier in 3.x to compile our ref/ assemblies against the 3.1.0 ref/ assemblies from outside aspnetcore. We only need to restrict ourselves to the 3.1.0 versions of these packages when building ref/ projects and Microsoft.AspNetCore.App.Ref. Further, we were already in 3.x servicing when we found the problem which made testing simpler.

With this change, we're building the ref/ and implementation projects with the same `@(Reference)` items and no longer have separate ref/ projects. That makes testing against the 5.0.x packages more complicated and gnarly to test.

Reply:
Oops, "ref/ and implementation **assemblies**"